standard/flags.lua

```diff
@@ -0,0 +1,171 @@
+---
+-- @Liquipedia
+-- wiki=commons
+-- page=Module:Flags
+--
+-- Please see https://github.com/Liquipedia/Lua-Modules to contribute
+--
+
+local MasterData = mw.loadData('Module:Flags/MasterData')
+local Template = require('Module:Template')
+local Logic = require('Module:Logic')
+local Table = require('Module:Table')
+local String = require('Module:StringUtils')
+local FnUtil = require('Module:FnUtil')
+
+local Flags = {}
+
+-- Template functions (legacy)
+function Flags.Flag(frame)
+	return Flags._Flag(frame.args[1])
+end
+function Flags.FlagNoLink(frame)
+	return Flags._FlagNoLink(frame.args[1])
+end
+
+--legacy functions
+function Flags._Flag(name)
+	return Flags.Icon(name)
+end
+function Flags._FlagNoLink(name)
+	return Flags.Icon({flag = name, noLink = true})
+end
+function Flags._CountryName(name)
+	return Flags.CountryName(name)
+end
+function Flags.flag(name)
+	return Flags.Icon(name)
+end
+function Flags.flagNoLink(name)
+	return Flags.Icon({flag = name, noLink = true})
+end
+
+-- Returns a flag
+--[[
+supported args are:
+flag	- name, flag code, or alias of the Flag
+noLink	- boolean that decides if the flag should link or not
```

Comment:
While we're at it, we can make this `shouldLink` (so inverted), that's easier to parse

Reply:
`showLink` works too I think

Reply:
should we make it `true` by default, i.e. something like
```lua
shouldLink = Logic.readBoolOrNil(shouldLink) ~= false
```
or should we have it as false by default?

Reply:
`true` by default tracks existing usage of `{{Flag/xx}}` and `Flags._Flag('xx')` (Using `{{FlagNoLink/xx}}` requires going a bit out of the way). 

However it might be good setting the default to `false` as a policy change. I don't think flags links are very useful. Most people clicking on flags probably did so by accident. Flags used by match2 are always nolink.

Reply:
> shouldLink = Logic.readBoolOrNil(shouldLink) ~= false

you can also use `Logic.nilOr(Logic.readBoolOrNil(shouldLink), true)` which makes it more clear that `true` is the default value by avoiding the negation

Reply:
i will start adjusting the calls in modules tomorrow 
will set true/false for both so that we can decide on the default next week